pkg/scalers/solace_scaler.go

```diff
@@ -266,40 +329,78 @@ func (s *SolaceScaler) getSolaceQueueMetricsFromSEMP(ctx context.Context) (Solac
 	//	Call Solace SEMP API
 	response, err := httpClient.Do(request)
 	if err != nil {
-		return SolaceMetricValues{}, fmt.Errorf("call to solace semp api failed: %w", err)
+		return SolaceMetricValues{}, fmt.Errorf("call to solace semp api failed: %s, url: %s", err.Error(), sempURL)
 	}
 	defer response.Body.Close()
 
 	// Check HTTP Status Code
 	if response.StatusCode < 200 || response.StatusCode > 299 {
-		sempError := fmt.Errorf("semp request http status code: %s - %s", strconv.Itoa(response.StatusCode), response.Status)
-		return SolaceMetricValues{}, sempError
+		return SolaceMetricValues{}, fmt.Errorf("semp request http status code: %d - %s, url: %s", response.StatusCode, response.Status, sempURL)
 	}
 
 	// Decode SEMP Response and Test
 	if err := json.NewDecoder(response.Body).Decode(&sempResponse); err != nil {
-		return SolaceMetricValues{}, fmt.Errorf("failed to read semp response body: %w", err)
+		return SolaceMetricValues{}, fmt.Errorf("failed to read semp response body: %s, url: %s", err.Error(), sempURL)
 	}
+
 	if sempResponse.Meta.ResponseCode < 200 || sempResponse.Meta.ResponseCode > 299 {
-		return SolaceMetricValues{}, fmt.Errorf("solace semp api returned error status: %d", sempResponse.Meta.ResponseCode)
+		return SolaceMetricValues{}, fmt.Errorf("solace semp api returned error status: %d, url: %s", sempResponse.Meta.ResponseCode, sempURL)
 	}
 
-	// Set Return Values
 	metricValues.msgCount = sempResponse.Collections.Msgs.Count
 	metricValues.msgSpoolUsage = sempResponse.Data.MsgSpoolUsage
 	metricValues.msgRcvRate = sempResponse.Data.MsgRcvRate
+
 	return metricValues, nil
 }
 
+// returns SolaceMetricValues struct populated from broker  SEMP endpoint
+func (s *SolaceScaler) getSolaceQueueMetricsFromSEMP(ctx context.Context) (SolaceMetricValues, error) {
+	var metricValues SolaceMetricValues
+	var errorList []string
+
+	s.stateLock.Lock()
+	defer s.stateLock.Unlock()
+
+	//	RETRIEVE METRICS FROM SOLACE SEMP API
+	for i := 0; i < len(s.metadata.endpointURLsList); i++ {
```

Comment:
Do we need to iterate over all the solace all the times? I mean, can't we just store the last index that has been successfully queried and use it until there is a failure? I don't now how solace works, so maybe this isn't possible, but I prefer to not query all the time all the endpoints.
With the current implementation, if the primary endpoint works, we will query just once, but if the primary endpoint fails, we will check it all the time and only after the failure, we will query the second one and maybe it makes sense if we want to prioritize the primary one